presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -331,10 +331,25 @@ public PlanWithProperties visitAggregation(AggregationNode node, StreamPreferred
             PlanWithProperties child = planAndEnforce(node.getSource(), childRequirements, childRequirements);
 
             List<VariableReferenceExpression> preGroupedSymbols = ImmutableList.of();
-            if (!LocalProperties.match(child.getProperties().getLocalProperties(), LocalProperties.grouped(groupingKeys)).get(0).isPresent()) {
+            // Logic in LocalProperties.match(localProperties, groupingKeys)
+            // 1. Extract the longest prefix of localProperties to a set that is a subset of groupingKeys
+            // 2. Iterate grouped-by keys and add the elements that's not in the set to the result
+            // Result would be a List of one element: Optional<GroupingProperty>, GroupingProperty would contain one/multiple elements from step 2
+            // Eg:
+            // [A, B] [(B, A)]   ->   List.of(Optional.empty())
+            // [A, B] [B]        ->   List.of(Optional.of(GroupingProperty(B)))
+            // [A, B] [A]        ->   List.of(Optional.empty())
+            // [A, B] [A, C]     ->   List.of(Optional.of(GroupingProperty(C)))
```

Comment:
The match function seems very powerful. :) Very clean solution. Thanks for figuring it out.

Could you please add another example for
[A, B] [D, A, C] (basically the first group by column is not part of the sort by columns, this case is not included in any of the above examples)
In this case will it return [D, C] and we can set A in the pre-grouped columns?
Please add a test for this case as well.